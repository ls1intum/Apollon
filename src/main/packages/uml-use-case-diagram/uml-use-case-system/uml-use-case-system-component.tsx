import React, { SFC } from 'react';
import { Text } from '../../../components/controls/text/text';
import { UMLUseCaseSystem } from './uml-use-case-system';

export const UMLUseCaseSystemComponent: SFC<Props> = ({ element, children }) => (
  <g>
    <rect width="100%" height="100%" stroke={element.strokeColor || 'black'} />
    <Text fill={element.textColor} y={16}>
      {element.name}
    </Text>
    {children}
  </g>
);

interface Props {
  element: UMLUseCaseSystem;
}
