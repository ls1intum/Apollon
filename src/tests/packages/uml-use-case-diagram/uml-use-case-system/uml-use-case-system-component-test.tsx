import * as React from 'react';
import { wrappedRender } from '../../../test-utils/render';
import { UMLUseCaseSystem } from '../../../../main/packages/uml-use-case-diagram/uml-use-case-system/uml-use-case-system';
import { UMLUseCaseSystemComponent } from '../../../../main/packages/uml-use-case-diagram/uml-use-case-system/uml-use-case-system-component';

it('render the uml-use-case-component', () => {
  const useCaseSystem: UMLUseCaseSystem = new UMLUseCaseSystem({ name: 'TestUseCaseComponent' });
  const { getByText, baseElement } = wrappedRender(
    <svg>
      <UMLUseCaseSystemComponent element={useCaseSystem} scale={1.0} />
    </svg>,
  );
  expect(getByText(useCaseSystem.name)).toBeInTheDocument();
  expect(baseElement).toMatchSnapshot();
});
