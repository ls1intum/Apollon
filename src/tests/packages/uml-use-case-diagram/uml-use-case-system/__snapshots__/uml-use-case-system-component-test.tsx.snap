// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the uml-use-case-component 1`] = `
<body>
  <div>
    <svg>
      <g>
        <rect
          height="100%"
          stroke="black"
          width="100%"
        />
        <text
          dominant-baseline="middle"
          font-weight="bold"
          pointer-events="none"
          text-anchor="middle"
          x="50%"
          y="16"
        >
          TestUseCaseComponent
        </text>
      </g>
    </svg>
  </div>
</body>
`;
